src/middlewared/middlewared/plugins/reporting/plugins_freebsd.py

```diff
@@ -213,7 +213,10 @@ class UPSBase(object):
     plugin = 'nut'
 
     def get_identifiers(self):
-        ups_identifier = self.middleware.call_sync('ups.config')['identifier']
+        ups_config = self.middleware.call_sync('ups.config')
+        ups_identifier = ups_config['identifier']
+        if ups_config['mode'] == 'SLAVE':
+            self._base_path = os.path.join(RRD_BASE_DIR_PATH, ups_config['remotehost'])
```

Comment:
Should we do this in the constructor instead?

Also RRDBase constructor has this
```
        self._base_path = RRD_BASE_PATH
        self.base_path = os.path.join(self._base_path, self.plugin)
```

Maybe we should just turn them into properties?